bn_mp_set_double.c

```diff
@@ -11,8 +11,21 @@
  *
  * SPDX-License-Identifier: Unlicense
  */
+#include <float.h>
+/* Minimum information needed */
+#if ( (defined DBL_MAX) && (FLT_RADIX == 2) )
 
-#if defined(__STDC_IEC_559__) || defined(__GCC_IEC_559)
+/* We can use a faster method if we have an IEEE compliant machine and a working stdint.h */
+#if (((defined __STDC_IEC_559__) || (defined __GCC_IEC_559)) && (defined UINT64_MAX))
+
+/* To avoid "magic numbers". Some people don't like them, even if their meaning is obvious */
+/* Bits in mantissa without the implied bit  */
+#define SIGNIFICANT (DBL_MANT_DIG - 1)
```

Comment:
@czurnieden It would be nice to avoid unqualified macros (even if private). I think this is better for clarity but there is also a technical reason. I am using tommath in a amalgamated form embedded in a project. This works perfectly and allows cross module optimization (without fancy LTO). This ease of embedding is one of the reasons why I like tommath so much :)  I haven't looked around the codebase but it would be nice to get rid of all unqualified identifiers. What do you think?

Ping @sjaeckel

Reply:
I kept the official macros from `float.h` (So instead of using `SIGNIFICANT` I used `(DBL_MANT_DIG - 1)`); `PAYLOAD`  and `EXPONENT_BIT` (in `mp_set_long_double`) have been reset to literals already. If you want *all* of your literals back (for easier c&p) drop me a note.

But I would not really like it, to be honest, the all-literals approach, that is.